### ISSUE TITLE
Some warnings appear even if from generated code

### ISSUE THREAD

**Issue by MisinformedDNA** [dotnet/roslyn-analyzers#1836]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.6.2

#### Diagnostic ID

[CA3075](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca3075-insecure-dtd-processing?view=vs-2015)

#### Repro steps

1. Create a .NET 4.6 Console app
1. Add the following to a new file

```csharp
// <auto-generated>
using System.IO;
using System.Xml.Schema;

namespace AutoGeneratedIssue
{
    public class CA3075_Generated
    {
        public XmlSchema Test
        {
            get
            {
                var src = "";
                TextReader tr = new StreamReader(src);
                XmlSchema schema = XmlSchema.Read(tr, null); // warn
                return schema;
            }
        }
    }
}
```

3. Add FxCopAnalyzer
4. Build

#### Expected behavior

No warnings are produced

#### Actual behavior

I still see the warning.

> CA3075	Unsafe overload of 'Read' method	AutoGeneratedIssue

#### Repro Project

Here is a [repro project](https://github.com/dotnet/roslyn-analyzers/files/2420410/AutoGeneratedIssue.zip) that has 4 classes. Two of them cause CA3075, one with "auto-generated" and the other without. And then as a control, the same thing with CA1058.

![image](https://user-images.githubusercontent.com/1784452/46090555-9848a700-c176-11e8-9a04-f3cb1b797358.png)

Notice that CA1058_Generated does not produce a warning, but CA3075_Generated does.

**Comment by mavasani** [comment]
@dotpaul Security analyzer needs to be updated to configure generated code analysis, for example: https://github.com/dotnet/roslyn-analyzers/blob/master/src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/PropertiesShouldNotReturnArrays.cs#L39.

**Comment by mavasani** [comment]
Ah, though I see it is already explicitly configured to analyze and report diagnostics in generated code: https://github.com/dotnet/roslyn-analyzers/blob/32d8f1e397439035f0ecb5f61a9e672225f0ecdb/src/Microsoft.NetFramework.Analyzers/Core/DoNotUseInsecureDtdProcessing.cs#L53

```
            // Security analyzer - analyze and report diagnostics in generated code.
            analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
```

**Comment by mavasani** [comment]
@MisinformedDNA I feel this might be by design. As this a security analyzer, we are erring on side of reporting issues in generated code, and the end user would have to explicitly suppress the security diagnostics on generated code if they do not want to fix security issues in generated code. Let us know if you feel otherwise.

**Comment by MisinformedDNA** [comment]
For curiosity sake, why is this issue configured in the [.NET Framework analyzers](https://github.com/dotnet/roslyn-analyzers/blob/master/src/Microsoft.NetFramework.Analyzers/Microsoft.NetFramework.Analyzers.md#ca3075-insecure-dtd-processing-in-xml), but not in the [.NET Core analyzers](https://github.com/dotnet/roslyn-analyzers/blob/master/src/Microsoft.NetCore.Analyzers/Microsoft.NetCore.Analyzers.md#ca2101-specify-marshaling-for-pinvoke-string-arguments). Is it no longer a security issue in .NET Core?

**Comment by mavasani** [comment]
@dotpaul Is the rule/API applicable for .NET Core as well? If so, we might need to consolidate the rule implementation similar to https://github.com/dotnet/roslyn-analyzers/pull/1793. Re-opening to keep the bug on Paul's radar.

**Comment by dotpaul** [comment]
I'll need to investigate the applicability of the security XML rules to .NET Core and newer versions of .NET Framework.

**Comment by MisinformedDNA** [comment]
I created a ruleset and disabled CA3075, and the warning _still_ comes through. That isn't by design, is it?

**Comment by dotpaul** [comment]
@MisinformedDNA Can you please attach the ruleset file you created?

I tried your project, using the AllRulesDisabled.ruleset (from packages\Microsoft.CodeAnalysis.FxCopAnalysis.2.6.2\rulesets\AllRulesDisabled.ruleset) and didn't see any warnings.

**Comment by MisinformedDNA** [comment]
@dotpaul Of course it isn't repo-ing anymore. :)
